src/PKSim.Core/Model/CompoundPK.cs

```diff
@@ -1,37 +1,89 @@
+
+      public CompoundPK()
+      {
+         bioAvailabilities = new Cache<int, double?>();
+         ddiAucInf = new Cache<int, double?>();
+         ddiCMax = new Cache<int, double?>();
+      }
       public string CompoundName { get; set; }
-      
+
       /// <summary>
-      /// AUC_inf of plasma curve following a IV infustion of 15 min
+      /// AUC_inf of plasma curve following a IV infusion of 15 min
       /// </summary>
-      public double? AucIV { get; set; }
+      private Cache<int, double?> bioAvailabilities { get; set; }
 
       /// <summary>
       /// AUC_inf of plasma curve following an application where only <see cref="CompoundName"/> is applied
       /// </summary>
-      public double? AucDDI { get; set; }
+      private Cache<int, double?> ddiAucInf { get; set; }
+      private Cache<int, double?> ddiCMax { get; set; }
 
-      /// <summary>
-      /// Cmax of plasma curve following an application where only <see cref="CompoundName"/> is applied
-      /// </summary>
-      public double? CmaxDDI { get; set; }
+      private double? valueStoreFor(ICache<int, double?> allValues, int individualId)
+      {
+         if (!allValues.Contains(individualId))
+            allValues[individualId] = null;
+
+         return allValues[individualId];
+      }
+
+      public ICache<int, double?> AllBioAvailabilities => bioAvailabilities;
+      public ICache<int, double?> AllDDIAucInf => ddiAucInf;
+      public ICache<int, double?> AllDDICMax => ddiCMax;
+
+      public double? BioAvailabilityAucInfFor(int individualId)
+      {
+         return valueStoreFor(bioAvailabilities, individualId);
+      }
+
+      public double? DDIAucInfFor(int individualId)
+      {
+         return valueStoreFor(ddiAucInf, individualId);
+      }
 
-      public CompoundPK Clone()
+      public void AddDDICMax(int individualId, double? cMax)
       {
-         return new CompoundPK
-         {
-            AucDDI = AucDDI,
-            AucIV = AucIV,
-            CompoundName = CompoundName
-         };
+         ddiCMax[individualId] = cMax;
+      }
+
+      public void AddDDIAucInf(int individualId, double? aucInf)
+      {
+         ddiAucInf[individualId] = aucInf;
+      }
+      
+      public double? CMaxDDIFor(int individualId)
+      {
+         return valueStoreFor(ddiCMax, individualId);
+      }
+
+      public void AddBioavailability(int individualId, double? aucInf)
+      {
+         bioAvailabilities[individualId] = aucInf;
+      }
+
+      public void AddQuantityPKParameter(QuantityPKParameter quantityPKParameter)
+      {
+         _quantityPKParameters.Add(quantityPKParameter);
+      }
+
+      public ICache<string, QuantityPKParameter> QuantityPKParameters() => _quantityPKParameters;
+
+      public double? QuantityFor(int individualId, string parameterName)
+      {
+         var quantityParameter = _quantityPKParameters.FirstOrDefault(x => string.Equals(x.Name, parameterName));
```

Comment:
Why are you using firstOrDefault and not the fact that this is a cache? (If this is the serialization issue, verify if it contains and then use)

Reply:
no, that's because it was a list, then I changed it to Cache and didn't clean that up.